File: Day36/SquareRootUsingBinarySearch.js
function squareRoot(number) {
  let start = 0;
  let end = number;
  let mid = parseInt((start + end) / 2);
  let ans = -1;
  while (start <= end) {
    const sqr = mid * mid;

    if (sqr === number) {
      return mid;
    } else if (sqr > number) {
      end = mid - 1;
    } else if (sqr < number) {
      ans = mid;
      start = mid + 1;
    }

    mid = parseInt((start + end) / 2);
  }
  return ans;
}

console.log(squareRoot(4));

File: Day-36/SquareRootUsingBinarySearch.py
def squareRoot(num):
    start = 0
    end = num
    mid = (start+end)//2
    ans = -1

    while(start<=end):
        square = mid * mid
        if(square == num):
            return mid
        elif(square>num):
            end = mid -1
        else:
            ans = mid
            start = mid +1
        mid = (start+end)//2
    
    return ans